### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by AVyha**
image
image
image
image
image
image
image

### CODE REVIEW

File: templates/includes/pagination.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        <a href="?page={{ page_obj.next_page_number }}" class="page-link">next</a>
+      </li>
+    {% else %}
+      <li class="page_item">
+        <a href="" class="page-link">next</a>
+      </li>
+    {% endif %}
+  </ul>
+
+{% endif %}
```
Comment (mltdll) [inline templates/includes/pagination.html:26]: It's better to always end files with a newline character. You can read why this is the case [here](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline), in the top answer.

**Review by mltdll — CHANGES_REQUESTED** [review]

**Review by mltdll — APPROVED** [review]

File: taxi/views.py [hunk 2]
```diff
@@ -17,3 +18,30 @@ def index(request):
     }
 
     return render(request, "taxi/index.html", context=context)
+
+
+class ManufacturerListView(generic.ListView):
+    model = Manufacturer
+    paginate_by = 5
+    queryset = Manufacturer.objects.order_by("name")
+    context_object_name = "manufacturer_list"
```
Comment (Y-Havryliv) [inline taxi/views.py:27]: add `template_name` (it's default, you've already added `context_object_name`)

File: taxi/views.py [hunk 3]
```diff
@@ -17,3 +18,30 @@ def index(request):
...
+
+
+class DriverListView(generic.ListView):
+    model = Driver
+    paginate_by = 5
+
+
+class DriverDetailView(generic.DetailView):
+    model = Driver
+    queryset = Driver.objects.prefetch_related("cars")
```
Comment (Y-Havryliv) [inline taxi/views.py:47]: did you look at the number of queries? and does this `prefetch_related` work?

File: templates/taxi/car_list.html [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{% extends "base.html" %}
```
Comment (Y-Havryliv) [inline templates/taxi/car_list.html:1]: Create table with the information of each instance.

File: templates/taxi/car_list.html [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+
+{% block content %}
+  <h2>Cars list</h2>
+  {% for car in car_list %}
```
Comment (Y-Havryliv) [inline templates/taxi/car_list.html:5]: what if the `car_list` was empty?

File: templates/taxi/car_list.html [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+    <li><a href="{% url 'taxi:car-detail' pk=car.id %}">{{ car.model }}</a></li>
```
Comment (Y-Havryliv) [inline templates/taxi/car_list.html:6]: In the car list set a link on the id field that leads to the car detail page.

File: templates/taxi/driver_list.html [hunk 7]
```diff
@@ -0,0 +1,8 @@
+{% extends "base.html" %}
+
+{% block content %}
+  <h2>Drivers list</h2>
+  {% for driver in driver_list %}
```
Comment (Y-Havryliv) [inline templates/taxi/driver_list.html:5]: what if `driver_list` was empty?

**Review by Y-Havryliv — CHANGES_REQUESTED** [review]

**Comment by AVyha** [comment]
image
image
same with drivers

**Review by Y-Havryliv — CHANGES_REQUESTED** [review]
Create templates for manufacturer list, car list, driver list. In these templates:
Create table with the information of each instance.
In the car list set a link on the id field that leads to the car detail page.
In the driver list set a link on the username field that leads to the driver detail page.

add tables to all your list templates

**Comment by AVyha** [comment]
image
image
image
done

**Review by Y-Havryliv — APPROVED** [review]
Nice!

**Pull request closed without merging**